simulator/scheduler/scheduler.go

```diff
@@ -67,124 +64,65 @@ func NewSchedulerService(client clientset.Interface, restclientCfg *restclient.C
 	return &Service{clientset: client, restclientCfg: restclientCfg, initialSchedulerCfg: initCfg, sharedStore: sharedStore, simulatorPort: simulatorPort}
 }
 
-func (s *Service) RestartScheduler(cfg *configv1.KubeSchedulerConfiguration) error {
-	if s.disabled {
-		return xerrors.Errorf("an external scheduler is enabled: %w", ErrServiceDisabled)
+func restartContainer(ctx context.Context, cli *client.Client, cfg *configv1.KubeSchedulerConfiguration) error {
+	containers, err := cli.ContainerList(ctx, container.ListOptions{})
+	if err != nil {
+		return xerrors.Errorf("failed to get container list: %w", err)
 	}
-
-	s.ShutdownScheduler()
-
-	oldSchedulerCfg := s.currentSchedulerCfg
-	if err := s.StartScheduler(cfg); err != nil {
-		klog.Infof("failed to start scheduler: %v. restarting with old configuration", err)
-		if err2 := s.StartScheduler(oldSchedulerCfg); err2 != nil {
-			klog.Warningf("failed to start scheduler with old configuration: %v", err2)
-			return xerrors.Errorf("start scheduler: %w, restart scheduler with old configuration: %w", err, err2)
+	for _, c := range containers {
+		if c.Names[0] != "/simulator-scheduler" {
+			continue
 		}
-		return xerrors.Errorf("start scheduler: %w", err)
-	}
-	return nil
-}
-
-func (s *Service) ResetScheduler() error {
-	return s.RestartScheduler(s.initialSchedulerCfg.DeepCopy())
-}
-
-// StartScheduler starts scheduler.
-//
-//nolint:funlen,cyclop
-func (s *Service) StartScheduler(versionedcfg *configv1.KubeSchedulerConfiguration) (retErr error) {
-	clientSet := s.clientset
-	restConfig := s.restclientCfg
-	ctx, cancel := context.WithCancel(context.Background())
-	defer func() {
-		if retErr != nil {
-			cancel()
+		if err := simulatorschedconfig.UpdateSchedulerConfig(cfg); err != nil {
+			return xerrors.Errorf("read old scheduler.yaml: %w", err)
 		}
-	}()
-	informerFactory := scheduler.NewInformerFactory(clientSet, 0)
-	var dynInformerFactory dynamicinformer.DynamicSharedInformerFactory
-	if restConfig != nil {
-		dynClient := dynamic.NewForConfigOrDie(restConfig)
-		dynInformerFactory = dynamicinformer.NewFilteredDynamicSharedInformerFactory(dynClient, 0, v1.NamespaceAll, nil)
-	}
-	evtBroadcaster := events.NewBroadcaster(&events.EventSinkImpl{
-		Interface: clientSet.EventsV1(),
-	})
-	evtBroadcaster.StartRecordingToSink(ctx.Done())
-
-	s.currentSchedulerCfg = versionedcfg.DeepCopy()
```

Comment:
@sanposhiho 
When does the bug occur?
In my local, the settings window seems to open under normal conditions.

Reply:
Just run up the simulator and press ⚙️ .

```
sanposhiho@Kenseis-MacBook-Pro kube-scheduler-simulator % git status
On branch ft/use_debuggable_scheduler
Your branch is up to date with 'YamasouA/ft/use_debuggable_scheduler'.

nothing to commit, working tree clean
sanposhiho@Kenseis-MacBook-Pro kube-scheduler-simulator % make docker_build_server docker_build_scheduler
docker build -f simulator/cmd/simulator/Dockerfile -t simulator-server simulator
[+] Building 6.9s (17/17) FINISHED                                                                                                                                                                                                                                                                                                                           docker:desktop-linux
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                                                                                                                                         0.0s
 => => transferring dockerfile: 667B                                                                                                                                                                                                                                                                                                                                         0.0s
 => [internal] load metadata for docker.io/library/golang:1.22                                                                                                                                                                                                                                                                                                               0.7s
 => [internal] load metadata for docker.io/library/alpine:3.14.0                                                                                                                                                                                                                                                                                                             0.7s
 => [internal] load .dockerignore                                                                                                                                                                                                                                                                                                                                            0.0s
 => => transferring context: 100B                                                                                                                                                                                                                                                                                                                                            0.0s
 => [build-env 1/8] FROM docker.io/library/golang:1.22@sha256:0ca97f4ab335f4b284a5b8190980c7cdc21d320d529f2b643e8a8733a69bfb6b                                                                                                                                                                                                                                               0.0s
 => [stage-1 1/3] FROM docker.io/library/alpine:3.14.0@sha256:adab3844f497ab9171f070d4cae4114b5aec565ac772e2f2579405b78be67c96                                                                                                                                                                                                                                               0.0s
 => [internal] load build context                                                                                                                                                                                                                                                                                                                                            0.0s
 => => transferring context: 30.03kB                                                                                                                                                                                                                                                                                                                                         0.0s
 => CACHED [build-env 2/8] RUN go env -w GOCACHE=/go-cache                                                                                                                                                                                                                                                                                                                   0.0s
 => CACHED [build-env 3/8] RUN go env -w GOMODCACHE=/gomod-cache                                                                                                                                                                                                                                                                                                             0.0s
 => CACHED [build-env 4/8] WORKDIR /go/src/simulator                                                                                                                                                                                                                                                                                                                         0.0s
 => CACHED [build-env 5/8] COPY go.mod go.sum ./                                                                                                                                                                                                                                                                                                                             0.0s
 => CACHED [build-env 6/8] RUN --mount=type=cache,target=/gomod-cache go mod download                                                                                                                                                                                                                                                                                        0.0s
 => [build-env 7/8] COPY . .                                                                                                                                                                                                                                                                                                                                                 0.7s
 => [build-env 8/8] RUN --mount=type=cache,target=/gomod-cache --mount=type=cache,target=/go-cache     go build -v -o ./bin/simulator ./cmd/simulator/simulator.go                                                                                                                                                                                                           5.4s
 => CACHED [stage-1 2/3] COPY --from=build-env /go/src/simulator/bin/simulator /simulator                                                                                                                                                                                                                                                                                    0.0s
 => CACHED [stage-1 3/3] RUN chmod a+x /simulator                                                                                                                                                                                                                                                                                                                            0.0s
 => exporting to image                                                                                                                                                                                                                                                                                                                                                       0.0s
 => => exporting layers                                                                                                                                                                                                                                                                                                                                                      0.0s
 => => writing image sha256:e0ca00a24de6c43c46f34e7cdf7e240c3fc8908088aff761ebffc4ccc6f7c333                                                                                                                                                                                                                                                                                 0.0s
 => => naming to docker.io/library/simulator-server                                                                                                                                                                                                                                                                                                                          0.0s

View build details: docker-desktop://dashboard/build/desktop-linux/desktop-linux/yzbl1s1fgqxxb78b5tnkj4jqx

What's next:
    View a summary of image vulnerabilities and recommendations → docker scout quickview 
docker build -f simulator/cmd/scheduler/Dockerfile -t simulator-scheduler simulator
[+] Building 3.9s (17/17) FINISHED                                                                                                                                                                                                                                                                                                                           docker:desktop-linux
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                                                                                                                                         0.0s
 => => transferring dockerfile: 737B                                                                                                                                                                                                                                                                                                                                         0.0s
 => [internal] load metadata for docker.io/library/golang:1.22                                                                                                                                                                                                                                                                                                               0.8s
 => [internal] load metadata for docker.io/library/alpine:3.14.0                                                                                                                                                                                                                                                                                                             0.6s
 => [internal] load .dockerignore                                                                                                                                                                                                                                                                                                                                            0.0s
 => => transferring context: 100B                                                                                                                                                                                                                                                                                                                                            0.0s
 => CACHED [stage-1 1/3] FROM docker.io/library/alpine:3.14.0@sha256:adab3844f497ab9171f070d4cae4114b5aec565ac772e2f2579405b78be67c96                                                                                                                                                                                                                                        0.0s
 => [build-env 1/8] FROM docker.io/library/golang:1.22@sha256:0ca97f4ab335f4b284a5b8190980c7cdc21d320d529f2b643e8a8733a69bfb6b                                                                                                                                                                                                                                               0.0s
 => [internal] load build context                                                                                                                                                                                                                                                                                                                                            0.0s
 => => transferring context: 8.23kB                                                                                                                                                                                                                                                                                                                                          0.0s
 => CACHED [build-env 2/8] RUN go env -w GOCACHE=/go-cache                                                                                                                                                                                                                                                                                                                   0.0s
 => CACHED [build-env 3/8] RUN go env -w GOMODCACHE=/gomod-cache                                                                                                                                                                                                                                                                                                             0.0s
 => CACHED [build-env 4/8] WORKDIR /go/src/simulator                                                                                                                                                                                                                                                                                                                         0.0s
 => CACHED [build-env 5/8] COPY go.mod go.sum ./                                                                                                                                                                                                                                                                                                                             0.0s
 => CACHED [build-env 6/8] RUN --mount=type=cache,target=/gomod-cache go mod download                                                                                                                                                                                                                                                                                        0.0s
 => CACHED [build-env 7/8] COPY . .                                                                                                                                                                                                                                                                                                                                          0.0s
 => [build-env 8/8] RUN --mount=type=cache,target=/gomod-cache --mount=type=cache,target=/go-cache     go build -v -o ./bin/scheduler ./cmd/scheduler/scheduler.go                                                                                                                                                                                                           2.4s
 => [stage-1 2/3] COPY --from=build-env /go/src/simulator/bin/scheduler /scheduler                                                                                                                                                                                                                                                                                           0.1s 
 => [stage-1 3/3] RUN chmod a+x /scheduler                                                                                                                                                                                                                                                                                                                                   0.3s 
 => exporting to image                                                                                                                                                                                                                                                                                                                                                       0.1s
 => => exporting layers                                                                                                                                                                                                                                                                                                                                                      0.1s
 => => writing image sha256:8f9101999a18bdd39cc707fe78ae27df6852e9ecc325f6f907a865f1a1e1b1d9                                                                                                                                                                                                                                                                                 0.0s
 => => naming to docker.io/library/simulator-scheduler                                                                                                                                                                                                                                                                                                                       0.0s

View build details: docker-desktop://dashboard/build/desktop-linux/desktop-linux/c2vrkfthw2hi4k96xm5gkajnv

What's next:
    View a summary of image vulnerabilities and recommendations → docker scout quickview 
sanposhiho@Kenseis-MacBook-Pro kube-scheduler-simulator % make docker_up_local
docker compose -f compose.yml -f compose.local.yml up -d
WARN[0000] /Users/sanposhiho/workspace/kube-scheduler-simulator/compose.local.yml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion 
[+] Running 9/9
 ✔ Network kube-scheduler-simulator_default                     Created                                                                                                                                                                                                                                                                                                      0.0s 
 ✔ Network kube-scheduler-simulator_simulator-internal-network  Created                                                                                                                                                                                                                                                                                                      0.0s 
 ✔ Volume "kube-scheduler-simulator_conf"                       Created                                                                                                                                                                                                                                                                                                      0.0s 
 ✔ Volume "kube-scheduler-simulator_simulator-etcd-data"        Created                                                                                                                                                                                                                                                                                                      0.0s 
 ✔ Container simulator-server                                   Started                                                                                                                                                                                                                                                                                                      0.4s 
 ✔ Container simulator-frontend                                 Started                                                                                                                                                                                                                                                                                                      0.4s 
 ✔ Container simulator-cluster                                  Started                                                                                                                                                                                                                                                                                                      0.4s 
 ✔ Container kube-scheduler-simulator-init-container-1          Started                                                                                                                                                                                                                                                                                                      0.4s 
 ✔ Container simulator-scheduler                                Started  
```

https://github.com/user-attachments/assets/cb1b6bc8-867c-4101-b507-3db254ba2425

Reply:
@sanposhiho 
I'm so sorry....
I fix this bug